shell/platform/android/io/flutter/app/FlutterPluginRegistry.java

```diff
@@ -238,5 +238,11 @@ public boolean onViewDestroy(FlutterNativeView view) {
 
     public void destroy() {
         mPlatformViewsController.onFlutterViewDestroyed();
+
+        mRequestPermissionsResultListeners.clear();
+        mActivityResultListeners.clear();
+        mNewIntentListeners.clear();
+        mUserLeaveHintListeners.clear();
+        mViewDestroyListeners.clear();
```

Comment:
Are these listeners already being unregistered elsewhere, or should this `destory()` call be handling that too?

It looks like they're registered before they're given to this class, so I think it makes sense for them to be unregistered elsewhere too. Just not 100% sure. /cc @matthew-carroll.

Reply:
I don't remember much about this because it's the old embedding, but here is the call stack that leads to this destruction:

`FlutterActivityDelegate#onDestroy()`: https://github.com/flutter/engine/blob/master/shell/platform/android/io/flutter/app/FlutterActivityDelegate.java#L240

`FlutterView#destroy()`: https://github.com/flutter/engine/blob/master/shell/platform/android/io/flutter/view/FlutterView.java#L423

`FlutterNativeView#destroy()`: https://github.com/flutter/engine/blob/master/shell/platform/android/io/flutter/view/FlutterNativeView.java#L66

`FlutterPluginRegistry#destroy()`: https://github.com/flutter/engine/blob/master/shell/platform/android/io/flutter/app/FlutterPluginRegistry.java#L239

You'll notice that this execution path only begins in `FlutterActivityDelegate` if no plugins are interested in running without a UI. That would have been my primary concern with this approach, but I *think* it's handled appropriately based on this call stack. So I'd say it's an LGTM if we can figure out how to write a test...